pkg/store/consul/consulutil/ops.go

```diff
@@ -96,6 +96,42 @@ func List(
 	}
 }
 
+type ConsulKeyser interface {
+	Keys(prefix, separator string, q *api.QueryOptions) ([]string, *api.QueryMeta, error)
+}
+
+type keysReply struct {
+	keys      []string
+	queryMeta *api.QueryMeta
+	err       error
+}
+
+// SafeKeys performs a KV Keys operation that can be canceled. When the "done"
+// channel is closed, CanceledError will be immediately returned. (The HTTP RPC
+// can't be canceled, but it will be ignored.) Errors from Consul will be
+// wrapped in a KVError value.
+func SafeKeys(
```

Comment:
the other ones are called `Safe` also, i agree with you that that word doesn't make sense to me